libsolidity/analysis/TypeChecker.cpp

```diff
@@ -226,43 +226,36 @@ TypePointers TypeChecker::typeCheckMetaTypeFunctionAndRetrieveReturnType(Functio
 {
 	vector<ASTPointer<Expression const>> arguments = _functionCall.arguments();
 	if (arguments.size() != 1)
-	{
-		m_errorReporter.typeError(
+		m_errorReporter.fatalTypeError(
 			8885_error,
 			_functionCall.location(),
 			"This function takes one argument, but " +
 			toString(arguments.size()) +
 			" were provided."
 		);
-		return {};
-	}
 	TypePointer firstArgType = type(*arguments.front());
 
 	bool wrongType = false;
 	if (firstArgType->category() == Type::Category::TypeType)
 	{
 		TypeType const* typeTypePtr = dynamic_cast<TypeType const*>(firstArgType);
 		Type::Category typeCategory = typeTypePtr->actualType()->category();
-		if (
-			typeCategory != Type::Category::Contract &&
-			typeCategory != Type::Category::Integer
-		)
+		if (auto const* contractType = dynamic_cast<ContractType const*>(typeTypePtr->actualType()))
+			wrongType = contractType->isSuper();
+		else if (typeCategory != Type::Category::Integer)
 			wrongType = true;
 	}
 	else
 		wrongType = true;
 
 	if (wrongType)
-	{
-		m_errorReporter.typeError(
+		m_errorReporter.fatalTypeError(
```

Comment:
Why is this converted into fatal?

Reply:
If this is not a fatal error, subsequent errors will be wrong because this returns a 'void' type. TypeChecker usually operates in a way that it either assigns the right type or it causes a fatal error.